src/expr.cpp

```diff
@@ -6177,56 +6236,61 @@ int ConstExpr::EstimateCost() const { return 0; }
 
 ConstExpr *ConstExpr::Instantiate(TemplateInstantiation &templInst) const { return new ConstExpr(this, pos); }
 
-void ConstExpr::Print(Indent &indent) const {
-    indent.Print("ConstExpr", pos);
-
-    printf("[%s] (", GetType()->GetString().c_str());
+std::string ConstExpr::GetValuesAsStr(const std::string &separator) const {
+    std::stringstream result;
     for (int i = 0; i < Count(); ++i) {
+        if (i != 0) {
+            result << separator;
+        }
         switch (getBasicType()) {
         case AtomicType::TYPE_BOOL:
-            printf("%s", boolVal[i] ? "true" : "false");
+            result << (boolVal[i] ? "true" : "false");
             break;
         case AtomicType::TYPE_INT8:
-            printf("%d", (int)int8Val[i]);
+            result << static_cast<int>(int8Val[i]);
             break;
         case AtomicType::TYPE_UINT8:
-            printf("%u", (int)uint8Val[i]);
+            result << static_cast<unsigned int>(uint8Val[i]);
             break;
         case AtomicType::TYPE_INT16:
-            printf("%d", (int)int16Val[i]);
+            result << static_cast<int>(int16Val[i]);
             break;
         case AtomicType::TYPE_UINT16:
-            printf("%u", (int)uint16Val[i]);
+            result << static_cast<unsigned int>(uint16Val[i]);
             break;
         case AtomicType::TYPE_INT32:
-            printf("%d", int32Val[i]);
+            result << int32Val[i];
             break;
         case AtomicType::TYPE_UINT32:
-            printf("%u", uint32Val[i]);
+            result << uint32Val[i];
             break;
         case AtomicType::TYPE_INT64:
-            printf("%" PRId64, int64Val[i]);
+            result << int64Val[i];
             break;
         case AtomicType::TYPE_UINT64:
-            printf("%" PRIu64, uint64Val[i]);
+            result << uint64Val[i];
             break;
-        case AtomicType::TYPE_FLOAT16: {
-            llvm::APFloat V(fpVal[i]);
-            printf("%f", V.convertToFloat());
+        case AtomicType::TYPE_FLOAT16:
+            result << std::to_string(llvm::APFloat(fpVal[i]).convertToFloat());
             break;
-        }
         case AtomicType::TYPE_FLOAT:
-            printf("%f", fpVal[i].convertToFloat());
+            result << std::to_string(fpVal[i].convertToFloat());
```

Comment:
Why `float16` and `float` cases are different? I looks like it's not intentional.

Reply:
Yes, it was not intentional. Fixed.